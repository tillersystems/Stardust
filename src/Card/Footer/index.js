import styled from 'styled-components';
import { transparentize } from 'polished';

/**
 * Footer
 *
 * This component is in charge of display
 * the footer of Card component
 *
 *
 * @return {jsx}
 */
const Footer = styled.div`
  box-shadow: 0 -1px 2px 0 ${({ theme: { palette } }) => transparentize(0.96, palette.black)};
  border-top: solid 1px ${({ theme: { palette } }) => palette.veryLightBlue};

  display: flex;
  align-items: center;
  justify-content: center;

  height: 4.9rem;
  padding: 0 2rem;
`;

export default Footer;
