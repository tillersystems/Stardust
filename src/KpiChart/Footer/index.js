import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const KpiChartFooter = ({ children, className }) => <div className={className}>{children}</div>;

/**
 * PropTypes Validation
 */
const { node, string } = PropTypes;
KpiChartFooter.propTypes = {
  children: node,
  className: string,
};

/**
 * Default props
 */
KpiChartFooter.defaultProps = {
  children: null,
  className: '',
};

export default styled(KpiChartFooter)`
  position: absolute;
  bottom: 1.5rem;
  left: 1rem;

  button {
    color: ${({ theme: { palette } }) => palette.primary.default};
  }
`;
