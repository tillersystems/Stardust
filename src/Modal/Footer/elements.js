import styled from 'styled-components';

export const Wrapper = styled.div`
  margin-left: ${({ theme: { dimensions } }) => dimensions.small};

  &:first-child {
    margin-left: 0;
  }

  &:last-child {
    margin-right: 0;
  }
`;
