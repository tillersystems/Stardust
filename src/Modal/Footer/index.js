import React, { Children } from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

import { Wrapper } from './elements';

const ModalFooter = ({ children, className, alignment }) => (
  <div className={className} alignment={alignment}>
    {Children.map(children, (child, index) => (
      <Wrapper key={index}>{child}</Wrapper>
    ))}
  </div>
);

/**
 * PropTypes Validation
 */
const { node, string } = PropTypes;
ModalFooter.propTypes = {
  children: node,
  className: string,
  alignment: string,
};

/**
 * Default props
 */
ModalFooter.defaultProps = {
  children: null,
  className: '',
  alignment: 'left',
};

export default styled(ModalFooter)`
  height: 6rem;
  width: 100%;
  display: flex;
  align-items: center;

  ${({ alignment }) =>
    alignment === 'left' &&
    css`
      justify-content: flex-start;
    `};

  ${({ alignment }) =>
    alignment === 'right' &&
    css`
      justify-content: flex-end;
    `};

  ${({ alignment }) =>
    alignment === 'center' &&
    css`
      justify-content: center;
    `};

  ${({ alignment }) =>
    alignment === 'spaced' &&
    css`
      justify-content: space-between;
    `};
`;
