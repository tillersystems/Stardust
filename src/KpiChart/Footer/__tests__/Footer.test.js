import React from 'react';

import Footer from '..';

describe('<KpiChart.Footer />', () => {
  test('should render without a problem', () => {
    const { container } = render(<Footer>Footer</Footer>);

    expect(container.firstChild).toMatchSnapshot();
  });
});
