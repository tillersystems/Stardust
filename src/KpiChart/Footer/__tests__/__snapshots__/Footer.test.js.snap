// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<KpiChart.Footer /> should render without a problem 1`] = `
.c0 {
  position: absolute;
  bottom: 1.5rem;
  left: 1rem;
}

.c0 button {
  color: hsl(200,74%,46%);
}

<div
  class="c0"
>
  Footer
</div>
`;
